src/System.Security.Cryptography.X509Certificates/ref/System.Security.Cryptography.X509Certificates.cs

```diff
@@ -166,7 +166,9 @@ public partial class X509Certificate : System.IDisposable, System.Runtime.Serial
         public virtual byte[] Export(System.Security.Cryptography.X509Certificates.X509ContentType contentType, string password) { throw null; }
         protected static string FormatDate(System.DateTime date) { throw null; }
         public virtual byte[] GetCertHash() { throw null; }
+        public virtual byte[] GetCertHash(HashAlgorithmName hashAlgorithm) { throw null; }
```

Comment:
Should we be thinking about a span-based try version of this?

Reply:
You could, but since different hashes have different sizes, it would probably be kind of ugly to use and since we don't have a Span-based hashing API, it would likely take some rewriting of our hash infrastructure. I guess we could do something like return a ReadOnlySpan and cache the underlying value, but I'm not certain there's a lot to gain from any of that.

Reply:
```C#
        public virtual bool TryGetCertHash(
            HashAlgorithmName hashAlgorithm,
            Span<byte> destination,
            out int bytesWritten) => throw null;
```
?

I feel like we've been leaving the verb as Get when the original verb was Get.

If that's the name you'd expect (@terrajobst said it sounded right to him) I've got it written now, just need to add some tests for it :smile:.